### PR TITLE
[maintenance] use standard ```_is_numpy_namespace``` for numpy namespace checking in PCA

### PR DESCRIPTION
**Pull request by icfaust**
## Description

```_is_numpy_namespace``` was introduced in #1861 which is used for overall numpy namespace checking.  This swaps a custom check with our standard one in order to reduce maintenance and make general conformance across the codebase.  This was also done in ```IncrementalEmpericalCovariance```.

No performance benchmarks necessary.
---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

### CODE REVIEW

**Comment by david-cortes-intel** [comment]
@icfaust What would happen in this case if the input falls into numpy namespace, but the means/components are not on CPU?

**Comment by icfaust** [comment]
> @icfaust What would happen in this case if the input falls into numpy namespace, but the means/components are not on CPU?

Interesting question.  The means and components would only be non numpy arrays currently if they successfully completed a PCA fit using array_api_dispatch with sklearn's fit. It would fail, but this would conform to what would occur with array_api_dispatch using sklearn's PCA, which means the failure is conforming to sklearn's failure.

However, our return values for array_api_dispatch using the onedal fit with non-numpy inputs should also create non-numpy array attributes, but currently only returns non-numpy arrays and sets numpy attributes. This is violation with sklearn array_api conformance, and when we change that, this would become a problem here.

**Review by david-cortes-intel — APPROVED** [review]
Ok, guess that's not in the scope of this PR.
